File: ruby/benchmark.rb
require_relative 'cpu_benchmark'
require_relative 'io_benchmark'

def main
  task_count = 10 # default
  is_io_mode = false
  
  # Parse command line arguments
  ARGV.each do |arg|
    if arg == 'io'
      is_io_mode = true
    elsif arg.match?(/^\d+$/)
      task_count = arg.to_i
    end
  end
  
  if is_io_mode
    run_io_benchmark(task_count)
  else
    run_cpu_benchmark(task_count)
  end
end

main

File: python/benchmark.py
#!/usr/bin/env python3
import sys
from cpu_benchmark import run_cpu_benchmark
from io_benchmark import run_io_benchmark

def main():
    # Parse command line arguments
    task_count = 10  # default
    is_io_mode = False
    
    for arg in sys.argv[1:]:
        if arg == 'io':
            is_io_mode = True
        elif arg.isdigit():
            task_count = int(arg)
    
    if is_io_mode:
        run_io_benchmark(task_count)
    else:
        run_cpu_benchmark(task_count)

if __name__ == "__main__":
    main()